pkgs/build-support/fetchsrht/default.nix

```diff
@@ -0,0 +1,26 @@
+{ fetchzip, lib }:
+
+{ owner, repo, rev
+, domain ? "sr.ht"
+, vc ? "git"
+, name ? "source"
+, ... # For hash agility
+} @ args:
+
+with lib;
+
+if !(elem vc [ "git" "hg" ])
+  then throw "fetchFromSourcehut: Version control ${vc} is not supported"
```

Comment:
IDK if there is a better way to do it

Reply:
I think we should assign a `type` of a repository and the `baseDomain` part independently. A subdomain could be set to arbitrary values (i.e. mercurial.example.org).

Reply:
`lib.assertOneOf`